calicoctl/calicoctl.go

```diff
@@ -100,6 +101,16 @@ Description:
 		command := arguments["<command>"].(string)
 		args := append([]string{command}, arguments["<args>"].([]string)...)
 
+		// Check for client/cluster version mismatch. If a mismatch occurs, check for
+		// --allow-version-mismatch arg to override/fail.
+		if err = commands.VersionMismatch(args); err != nil {
```

Comment:
Instead of always running this, it might be better to run it only when `--allow-version-mismatch` is not passed in

Reply:
Right, good point! Will do, thanks!